src/modules/project/pages1/projectView/views/rewards/hooks/useProjectRewardForm.ts

```diff
@@ -0,0 +1,366 @@
+/* eslint-disable complexity */
+import { yupResolver } from '@hookform/resolvers/yup'
+import { format } from 'date-fns'
+import { ChangeEvent, useCallback, useEffect, useState } from 'react'
+import { useForm } from 'react-hook-form'
+import { useNavigate } from 'react-router'
+import * as yup from 'yup'
+
+import { useBTCConverter } from '@/helpers'
+import { useProjectRewardsAPI } from '@/modules/project/API/useProjectRewardsAPI'
+import { useProjectAtom } from '@/modules/project/hooks/useProjectAtom'
+import { getPath } from '@/shared/constants'
+import { useModal } from '@/shared/hooks'
+import {
+  CreateProjectRewardInput,
+  PrivateCommentPrompt,
+  ProjectRewardFragment,
+  RewardCurrency,
+  Satoshis,
+  UpdateProjectRewardInput,
+  USDCents,
+  useProjectRewardCurrencyUpdateMutation,
+  useProjectRewardQuery,
+  useRewardCategoriesQuery,
+} from '@/types'
+import { useNotification } from '@/utils'
+
+type FormValues = Omit<ProjectRewardFragment, 'id' | 'sold'>
+
+const MAX_REWARD_IMAGES = 5
+
+const rewardFormSchema = (soldAmount: number) =>
+  yup.object().shape({
+    name: yup.string().required('Name is required').max(50, 'Name must be at most 50 characters long'),
+    description: yup.string().max(1200, 'Description must be at most 1200 characters long'),
+    shortDescription: yup.string().max(250, 'Short Description must be at most 250 characters long'),
+    cost: yup
+      .number()
+      .typeError('Price is required')
+      .required('Price is required')
+      .min(0.01, 'Price must be greater than 0'),
+    maxClaimable: yup.number().nullable(),
+    images: yup.array().of(yup.string()).max(MAX_REWARD_IMAGES, `Maximum ${MAX_REWARD_IMAGES} images allowed`),
+    hasShipping: yup.boolean(),
+    isAddon: yup.boolean(),
+    isHidden: yup.boolean(),
+    category: yup.string().nullable(),
+    preOrder: yup.boolean(),
+    rewardCurrency: yup.string(),
+    estimatedAvailabilityDate: yup.date().nullable(),
+    estimatedDeliveryInWeeks: yup.number().nullable().min(0, 'Delivery time must be greater than or equal to 0'),
+    confirmationMessage: yup.string().max(500, 'Confirmation message must be at most 500 characters long'),
+    privateCommentPrompts: yup.array().of(yup.string()),
+  })
```

Comment:
don't see a use of this `solidAmount`  inside this function

Reply:
Yes you're right, I tried to use it to make a proper validation for the `maxClaimable` condition that cannot be lower than the `soldAmount` but it didn't work as I expected - I insist in doing in yup to keep the whole validation consistent. 
Good catch!